tests/if_else/do_no_path.sy

```diff
@@ -0,0 +1,14 @@
+f :: fn a do
+    if a == 1 do
+        <!>
+    elif a == 2 do
+        <!>
+    end
+end
+
+start :: fn do
+    f' 3
+    f' 0
+end
+
+// error: type isn't explicitly set to `void`
```

Comment:
Not related to this pr, but...

Isn't this a bit weird. The function is declared with `f :: fn a do` and not `f :: fn a ->`, and since there is no need for the ret keyword you cannot know at a glance if the function actually returns something. Take this for instance:

```
g :: fn do
    1
end

f :: fn do
    g'
end

start :: fn do
    print' f'
end
```

If you look at the definition of `f` there is no clear indication that it actually returns a value rather than void.